### ISSUE TITLE
Last Airship Stage Bug

### ISSUE THREAD

**Issue by Xplode111** [HBroHei/NSMRWii#13]
Hey there, I found another bug while playing the game. 
Basically, the last airship stage (from world 8) is not programmed to work out of its original position or something, because when you randomize the game and beat Bowsy in that stage there should be an animation with the Princess Peach after the bossfight, but instead you appear in an empty void with just the background and then you die instantly. I report this as an issue because I had the bad luck to get to this stage in the world 1-2 with the randomizer, meaning I cannot progress anymore and I have to reroll the stages or something. 
I think something similar may happen with the last stage of world 8 (big castle), but I don't know, I haven't played that stage with randomized format.
I recently played a castle stage which was out of its position (specifically the goomba in the world 1) and something similar happened. After the bossfight you don't die, but there are the background and some blocks used for the animation, and you can't move or do anything. That means you can't progress anymore (again), it could have got worse if it wasn't a skippable stage.

**Comment by HBroHei** [comment]
Hi, after replacing 1-2 with vanilla 8-Airship, the cutscene would work fine, and I can progress as usual.

Can you send me your config.json so that I can replicate it again?

**Comment by Xplode111** [comment]
Yeah, it is attached to this message, and I could understand that the error happened because of the config file
Just because I tried with 3 different seeds, 2 of them casually switched 1-2 with 8-Airship (both produced the error I said), and this config file which was the last I created, replaces the goomba from world 1 with 3-Castle (it also produced the error, but at its way)
All you have to do to replicate the error is getting through the stages until you are in the part of world 1 where is a Goomba, enter into its stage and just beat it.
I also made a clip beating the koopaling (I don't remember his name lol) and the situation I'm reporting happening later if by some chance the game lets you pass the stage without any problem, but the file is too big and I want to wait for a response if you know what's happening and if you can get through that someway.
[config.json](https://github.com/HBroHei/NSMRWii/files/15430554/config.json)

**Comment by HBroHei** [comment]
Thank you very much for sending me the config.json file, and my sincere apologies for not able to replicate your seed since some other external factor would affect the seed result.
The external factor bug should be fixed in [this version](https://github.com/HBroHei/NSMRWii/releases/tag/beta1.2.6.1). Please download this version, and if you have encountered the situation you have mentioned, please send in the config.json file to me, so that I can debug with that file.

For your other questions:
- As 2-Castle have some hardcoded elements that wouldn't work with other levels, the config.json generator actually always ensure 2-Castle is not randomised. (You can still put it into the randomisation pool via [Advanced config.json editing](https://hbrohei.github.io/NSMRWii/generator/tut_config.html) though, but it may be too advanced for you and average users)
- You can post the clip on cloud storages / upload it to YouTube, although I imagine I would have a harder time figuring out what's wrong since I do not have access to the original game files.
- I am aware that the Enemy Ambush chests are broken. This is likely a side effect of "Star house chest randomisation", but also I will be fixing it in the future.
- Jellybeam (the jelleyfish enemy) does produce a white rectangle when the stage is not dark. However, I kept it in as I considered it a "challenge" for the player.

Again, apology for having you to basically find the bug again, but I hope I can identify the issue as soon as possible. (Although I would be busy the following week and not able to do any efficient debugging until next Sunday - but I still got some time on Monday)

**Comment by Xplode111** [comment]
Ok, I tried now with the new version of the randomizer and using the same config.json file I sent to you, but I still had that problem I said (even in 2-Castle, as usual)
Here I uploaded to Mediafire the clip I mentioned, and the problem is visible at the second 00:16, but I put there some movement to prove it's real. That happens in every stage that has a cutscene after beating it. If it is necessary, I could manage to send you the raw DATA folder and see what causes this bug. Thanks
(I made the video fast and I also don't know how to crop the Windows Taskbar from the vid, so don't pay much attention to it xD)
https://www.mediafire.com/file/hm06oirj90efe7l/clip.mp4/file
